shared/chat/conversation/search/index.js

```diff
@@ -1,82 +1,92 @@
 // @flow
 
 import * as React from 'react'
+import * as Types from '../../../constants/types/chat2'
 import * as Kb from '../../../common-adapters'
 import * as Styles from '../../../styles'
 import {formatTimeForMessages} from '../../../util/timestamp'
-import type {Props} from './index.types'
 
 const hitHeight = 30
 
+type SearchHit = {|
+  author: string,
+  summary: string,
+  timestamp: number,
+|}
+export type Props = {
+  hits: Array<SearchHit>,
+  loadSearchHit: number => void,
+  onCancel: () => void,
+  onSearch: string => void,
+  selfHide: () => void,
+  status: Types.ThreadSearchStatus,
+  style?: Styles.StylesCrossPlatform,
+}
+
 type State = {|
   selectedIndex: number,
 |}
 
-class ThreadSearch extends React.Component<Props, State> {
+class ThreadSearchBase extends React.Component<Props, State> {
```

Comment:
I think this should be a HOC, it can make a class to keep track of the state and pass down any of these fcns the desktop/mobile sides need as props.